Flashlight-VNC/src/com/flashlight/vnc/VNCClient.as

```diff
@@ -599,78 +609,64 @@ package com.flashlight.vnc
 					case Keyboard.F11  		: keysym = 0xFFC8; break;
 					case Keyboard.F12  		: keysym = 0xFFC9; break;
 					case 91               : keysym = 0xFFEB; break; // Windows key
-					case Keyboard.SHIFT 	:
-						keysym = 0xFFE1;
-						shiftKeyDown = event.type == flash.events.KeyboardEvent.KEY_DOWN;
-						break;
-					case Keyboard.CONTROL	: keysym = 0xFFE3; break;
 				}
+				if (keysym != 0) {
+					rfbWriter.writeKeyEvent(event.type == flash.events.KeyboardEvent.KEY_DOWN,keysym);
+				}
+				logger.info("<< onLocalKeyboardEvent()");
+			}
+		}
 
-				//logger.info("keysym "+keysym);
-
-				rfbWriter.writeKeyEvent(event.type == flash.events.KeyboardEvent.KEY_DOWN,keysym);
-
-				// prevent onTextInput() from firing
-				//event.stopImmediatePropagation();
+		private function sendChar(charToSend:Object):void {
+			if (charToSend.shifted == true) {
+				rfbWriter.writeKeyEvent(true, 0xFFE1, false);
+			}
+			if (charToSend.controled == true) {
+				rfbWriter.writeKeyEvent(true, 0xFFE3, false);
+			}
+			rfbWriter.writeKeyEvent(true, charToSend.code, true);
+			if (charToSend.shifted == true) {
+				rfbWriter.writeKeyEvent(false, 0xFFE1, false);
+			}
+			if (charToSend.controled == true) {
+				rfbWriter.writeKeyEvent(false, 0xFFE3, false);
+			}
```

Comment:
It just occurred to me (and really this is just out of interested) does our approach work with chorded commands like you find in emacs (e.g. Ctrl-Alt-A-Z)?